app/controllers/concerns/listings_toolkit.rb

```diff
@@ -4,14 +4,19 @@ module ListingsToolkit
 
   MANDATORY_FIELDS_FOR_UPDATE = %i[body_markdown title tag_list].freeze
 
-  def set_listing
+  included do
+    before_action :set_and_authorize_listing, only: %i[update]
+  end
+
+  def set_and_authorize_listing
     @listing = Listing.find(params[:id])
+    authorize @listing
   end
 
   def rate_limit?
     begin
       rate_limit!(:listing_creation)
-    rescue StandardError => e
+    rescue ::RateLimitChecker::LimitReached => e
```

Comment:
I felt like we should check for the correct error here. The method is narrowly scoped so we shouldn't just swallow unrelated errors.